File: service/router/admin/redeemCode.go
package admin

import (
	"sun-panel/api/api_v1"
	"sun-panel/api/api_v1/middleware"

	"github.com/gin-gonic/gin"
)

func InitRedeemCodeRouter(router *gin.RouterGroup) {
	api := api_v1.ApiGroupApp.ApiAdmin.RedeemCodeApi

	r := router.Group("redeemCode", middleware.LoginInterceptor, middleware.AdminInterceptor)
	{
		r.POST("getList", api.GetList)
		r.POST("create", api.Create)
		r.POST("setInvalid", api.SetInvalid)
	}

}

File: src/api/admin/redeemCode.ts
import { post } from '@/utils/request'

export function getList<T>(req: Admin.RedeemCode.GetListRequest) {
  return post<T>({
    url: '/admin/redeemCode/getList',
    data: req,
  })
}

export function create<T>(data: RedeemCode.CreateReq) {
  return post<T>({
    url: '/admin/redeemCode/create',
    data,
  })
}

export function setInvalid<T>(codes: string[]) {
  return post<T>({
    url: '/admin/redeemCode/setInvalid',
    data: { codes },
  })
}
